controllers/hcloudremediation_controller_test.go

```diff
@@ -0,0 +1,292 @@
+					DataSecretName: ptr.To("bootstrap-secret"),
+				},
+			},
+		}
+		Expect(testEnv.Create(ctx, capiMachine)).To(Succeed())
+
+		capiMachineKey = client.ObjectKey{Name: capiMachineName, Namespace: testNs.Name}
+
+		hetznerCluster = &infrav1.HetznerCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "hetzner-test1",
+				Namespace: testNs.Name,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: clusterv1.GroupVersion.String(),
+						Kind:       "Cluster",
+						Name:       capiCluster.Name,
+						UID:        capiCluster.UID,
+					},
+				},
+			},
+			Spec: getDefaultHetznerClusterSpec(),
+		}
+		Expect(testEnv.Create(ctx, hetznerCluster)).To(Succeed())
+
+		hcloudMachine = &infrav1.HCloudMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      hcloudMachineName,
+				Namespace: testNs.Name,
+				Labels: map[string]string{
+					clusterv1.ClusterNameLabel:             capiCluster.Name,
+					clusterv1.MachineControlPlaneNameLabel: "",
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: clusterv1.GroupVersion.String(),
+						Kind:       "Machine",
+						Name:       capiMachine.Name,
+						UID:        capiMachine.UID,
+					},
+				},
+			},
+			Spec: infrav1.HCloudMachineSpec{
+				ImageName:          "fedora-control-plane",
+				Type:               "cpx31",
+				PlacementGroupName: &defaultPlacementGroupName,
+			},
+		}
+		Expect(testEnv.Create(ctx, hcloudMachine)).To(Succeed())
+
+		hcloudMachineKey = client.ObjectKey{Name: hcloudMachineName, Namespace: testNs.Name}
+
+		hcloudRemediation = &infrav1.HCloudRemediation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "hcloud-remediation",
+				Namespace: testNs.Name,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "Machine",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       capiMachine.Name,
+						UID:        capiMachine.UID,
+					},
+				},
+			},
+			Spec: infrav1.HCloudRemediationSpec{
+				Strategy: &infrav1.RemediationStrategy{
+					Type:       "Reboot",
+					RetryLimit: 1,
+					Timeout:    &metav1.Duration{Duration: 1 * time.Second},
+				},
+			},
+		}
+
+		hcloudRemediationkey = client.ObjectKey{Namespace: testNs.Name, Name: "hcloud-remediation"}
+	})
+
+	AfterEach(func() {
+		Expect(testEnv.Cleanup(ctx, testNs, hcloudRemediation, hcloudMachine,
+			hetznerSecret, bootstrapSecret, hetznerCluster, capiMachine, capiCluster)).To(Succeed())
+	})
+
+	Context("Basic test", func() {
+		It("creates the hcloudRemediation successfully", func() {
+			Expect(testEnv.Create(ctx, hcloudRemediation)).To(Succeed())
+
+			Eventually(func() error {
+				return testEnv.Get(ctx, hcloudRemediationkey, hcloudRemediation)
+			}, timeout).Should(BeNil())
+		})
+
+		It("checks if hcloudRemediation objects has the HCloudTokenAvailableCondition condition", func() {
+			Expect(testEnv.Create(ctx, hcloudRemediation)).To(Succeed())
+
+			Eventually(func() bool {
```

Comment:
you can remove this

Reply:
I mean you can remove the "get" call because you do it anyway in isPresentAndTrue!